### ISSUE TITLE
parseFiles hangs when extending interface that has method returning type from webpack

### ISSUE THREAD

**Issue by malcolm-kee** [ccontrols/structured-types#9]
I tried this with other library which seems fine. Not sure if it's due to how webpack type is declared.

Reproduction: https://stackblitz.com/edit/node-h68dpt?file=src/type.ts

Code

```ts
import { Configuration } from 'webpack';

export interface FullAnimal {
  webpackConfig?: () => Configuration;
}

export interface NameOnly extends FullAnimal {}
```

**Comment by malcolm-kee** [comment]
Can't reproduce, let me try to find out...

**Comment by malcolm-kee** [comment]
Able to create a reproduction: https://stackblitz.com/edit/node-h68dpt?file=src/type.ts

**Comment by atanasster** [comment]
Thank you, I am able to reproduce and will let you know as soon as I can find a fix

**Comment by atanasster** [comment]
@malcolm-kee just released v `3.21.1` that should fix the issue. Can you please confirm so we can close it.

Thanks again for the detailed issue and reproduction example

**Comment by malcolm-kee** [comment]
Yes, works now. Thanks!
